packages/loopring_v3.js/src/request_processors/deposit_processor.ts

```diff
@@ -0,0 +1,71 @@
+import BN from "bn.js";
+import { Bitstream } from "../bitstream";
+import { Constants } from "../constants";
+import { Account, Block, Deposit, ExchangeState } from "../types";
+
+/**
+ * Processes deposit requests.
+ */
+export class DepositProcessor {
+  public static processBlock(state: ExchangeState, block: Block) {
+    const offset = 4 + 32 + 32 + 32 + 32;
+    const data = new Bitstream(block.data);
+    const startIdx = data.extractUint32(offset);
+    const length = data.extractUint32(offset + 4);
+
+    const deposits: Deposit[] = [];
+    for (let i = startIdx; i < startIdx + length; i++) {
+      const deposit = state.deposits[i];
+
+      this.processDeposit(state, deposit);
+
+      deposit.blockIdx = block.blockIdx;
+      deposit.requestIdx = state.processedRequests.length + i - startIdx;
```

Comment:
should we push this deposit into `state.processedRequests`, if so, maybe we can do:

```
state.processedRequests.push(deposit);
deposits.push(deposit);
deposit.requestIdx = state.processedRequest.length;
```

Reply:
I don't push them in `state.processedRequests` here because it's not needed when the block is replayed to restore the state in the case of reverts. So this is done in a level above in `ExchangeV3.processBlock` which knows if this is the initial processing of the block or a replay.